projects/typst/build.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -eu
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+cd tests/fuzz
+cargo +nightly-2023-11-03 fuzz build -O --debug-assertions
```

Comment:
Is `+nightly-2023-11-03` necessary, and if so, what for?

Reply:
cargo fuzz [requires nightly to instrument with -fsanitize=address](https://rust-fuzz.github.io/book/cargo-fuzz/setup.html?highlight=nightly#requirements). You won't see this on the other build.sh files as nightly is set as [the default in the rust images](https://github.com/google/oss-fuzz/blob/55650963e17693deedfe2ed86df2544cf124ab8b/infra/base-images/base-builder/install_rust.sh#L21). When building typst on the latest nightly compiler we ended up triggering multiple bugs in the compiler. There is a discussion [here](https://github.com/typst/typst/pull/2581#issuecomment-1799804816) about two of the bugs, it turns out that there where a couple more. After bisecting out different versions of nightly compiler I came up with a specific nightly version that builds typst just fine without any issues. Hence the very specific nightly date.

Reply:
It would be nice to add this as a comment, or even better, to provide links to the bugs so that the flag can be removed once they're fixed :)

Reply:
@silvergasp note that the nightly here is a different than in Typst's CI. I changed it to 2023-09-13 there before merging your PR because we also use that specific nightly version in other parts of our code base and I wanted to keep everything pinned to the same thing.

Reply:
I just tried the latest nightly. Everything seems to be building fine. @laurmaedje do you think its worth unpinning this?

Reply:
I tried this again with the latest nightly and it failed again. I think with the large API surface of typst building on latest/nightly is too hard. So I've opted to pin at the version that @laurmaedje has listed.